pandas/core/frame.py

```diff
@@ -7217,10 +7217,14 @@ def _join_compat(
             # join indexes only using concat
             if can_concat:
                 if how == "left":
-                    res = concat(frames, axis=1, join="outer", verify_integrity=True)
+                    res = concat(
+                        frames, axis=1, join="outer", verify_integrity=True, sort=sort
+                    )
                     return res.reindex(self.index, copy=False)
                 else:
-                    return concat(frames, axis=1, join=how, verify_integrity=True)
+                    return concat(
```

Comment:
I suspect that this branch is under-tested... Could you update your new test to also assert the result of the `concat` operation?